### PR TITLE
Bump pandas to 0.25.0; test updates

### LINKED ISSUE #1447: Bug in pandas assert_*_equal wrapper test causes other tests to pass when they should fail
**Describe the bug**
Using the 36-min environment, I get different test results in the iotools tests depending on whether I include `test_conftest.py` in the test run.

**To Reproduce**
```bash
$ conda env create --file ci/requirements-py36-min.yml -n pvlib36min
$ conda activate pvlib36min
$ pytest pvlib/tests/test_conftest.py pvlib/tests/iotools  # no failures
$ pytest pvlib/tests/iotools/  # some failures
```

**Expected behavior**
I expect to be able to run test modules individually and get the same results I'd get by running the entire test suite.

**Versions:**
 - ``pvlib.__version__``: master (`0.9.1+2.g8d0f863`)
 - ``pandas.__version__``: 0.22.0
 - python: 3.6.13

**Additional context**
The failures are mostly about whether dataframe columns and indexes are equal, so it might have something to do with the `assert_X_equal` definitions in `conftest.py`?  I also mentioned the psm3 failure here: https://github.com/pvlib/pvlib-python/pull/1374#pullrequestreview-948638981

### LINKED ISSUE #1449: Output of solarposition.sun_rise_set_transit_ephem depends on installed ephem version
**Describe the bug**
`pvlib.solarposition.sun_rise_set_transit_ephem` returns a different answer depending on what version of `ephem` is installed. I think the problem is that our wrapper assumes that ephem doesn't pay attention to timezone localization, so it converts the timestamp components to UTC but doesn't bother to change the timestamp's tzinfo:

https://github.com/pvlib/pvlib-python/blob/8d0f863da92739669e01ac4da4145e4547638b50/pvlib/solarposition.py#L577-L579

However, starting in `ephem==4.1.1` the timezone is no longer ignored ([ref](https://rhodesmill.org/pyephem/CHANGELOG.html#version-4-1-1-2021-november-27)), so the UTC offset is applied twice.  This can shift the timestamp into the next solar period and return the rise/set/transit for the wrong day. 

**To Reproduce**
See how the returned sunrise differs by ~24 hours (2019-01-01 vs 2019-01-02) here:

```python
import pandas as pd
import pvlib
times = pd.date_range('2019-01-01', freq='h', periods=1, tz='Etc/GMT+8')
out = pvlib.solarposition.sun_rise_set_transit_ephem(times, 40, -120)
print(out.T)
```

#### `ephem==4.0.0.1`:
```

               2019-01-01 00:00:00-08:00
sunrise 2019-01-01 07:21:28.793036-08:00
sunset  2019-01-01 16:45:50.959086-08:00
transit 2019-01-01 12:03:35.730674-08:00
```

#### `ephem==4.1.2`:
```
               2019-01-01 00:00:00-08:00
sunrise 2019-01-02 07:21:35.237404-08:00
sunset  2019-01-01 16:45:50.947472-08:00
transit 2019-01-01 12:03:35.728413-08:00
```

**Expected behavior**
pvlib should give ephem timestamps consistent with its input requirements.  Replacing the above manual utc offset subtraction (which leaves the original tzinfo in place) with `thetime.astimezone(pytz.UTC)` may be suitable for both old and new versions of ephem.  I don't ever use pytz and python datetimes so maybe there's a better alternative.

**Versions:**
 - ``pvlib.__version__``: 0.9.1
 - python: 3.7

**Additional context**
This difference would have popped up back in November when ephem 4.1.1 was released had it not been for #1447.  Here's an example failure: https://dev.azure.com/solararbiter/pvlib%20python/_build/results?buildId=6027&view=logs&j=e1592cb8-2816-5754-b393-3839a187d454&t=377c4fd6-97bd-5996-bc02-4d072a8786ea&l=2267

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Closes #1447 
 - [x] Closes #1449
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Initially taking the less aggressive approach of only bumping pandas to 0.25.0, but wouldn't mind going straight to 1.1.0 if that's what others would prefer.

### CODE REVIEW

**Review by kanderso-nrel — COMMENTED** [review]
Ready for review.  For context, this PR fixes #1447 and cleans up the various issues it was hiding. 

I opted to do a hodgepodge of fixes in a single PR so that we don't have broken tests on master.  Let me know if this is too annoying to review and I'll divide it into smaller PRs.

File: pvlib/solarposition.py [hunk 1]
```diff
@@ -574,9 +575,10 @@ def sun_rise_set_transit_ephem(times, latitude, longitude,
     trans = []
     for thetime in times:
         thetime = thetime.to_pydatetime()
-        # pyephem drops timezone when converting to its internal datetime
-        # format, so handle timezone explicitly here
-        obs.date = ephem.Date(thetime - thetime.utcoffset())
+        # older versions of pyephem ignore timezone when converting to its
+        # internal datetime format, so convert to UTC here to support
+        # all versions.  GH #1449
+        obs.date = ephem.Date(thetime.astimezone(pytz.UTC))
```
Comment (wholmgren) [inline pvlib/solarposition.py:581]:
```suggestion
        obs.date = ephem.Date(thetime.astimezone(datetime.timezone.UTC))
```

**Comment by kanderso-nrel** [comment]
Hmm, bare linux 3.8 and 3.9 have an interesting tracking.py failure: https://dev.azure.com/solararbiter/pvlib%20python/_build/results?buildId=6045&view=logs&j=88437082-9fce-53b3-ffed-ea022e86ea67&t=6b9e297c-f567-5c08-dab9-9c9bfc583be2&l=191

Maybe related to numpy's adoption of SIMD?  It doesn't seem to happen every time, which I think is consistent with it being SIMD-related, plus the only affected CI variants are what you'd expect for SIMD strangeness.  I will look into that before declaring victory here.

**Comment by kanderso-nrel** [comment]
Testing on the benchmark server (linux w/ new CPU, necessary for numpy SIMD to be active) confirms it's a small floating point rounding issue that goes away when SIMD is disabled.  So similar to #1191, I just added a call to `np.clip` to keep the argument of `arccos` from going out of range.

For reference, here's how to run pytest with SIMD disabled:
```bash
$ NPY_DISABLE_CPU_FEATURES="AVX512F,AVX512CD,AVX512VL,AVX512BW,AVX512DQ,AVX512_SKX" pytest pvlib/tests/test_tracking.py
```

**Pull request merged**